### PR TITLE
Update Go dependencies

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

Just some Go dependency updates

#### Changes

- Update our Go modules
- Fix import path for OpenCensus Prometheus exporter

#### Notes for Reviewers

@rvolosatovs do we actually still need the `replace github.com/golang/lint => golang.org/x/lint` and `replace github.com/testcontainers/testcontainer-go => github.com/testcontainers/testcontainers-go`?

### CODE REVIEW

**Comment by rvolosatovs** [comment]
The reason why we had these replacements was because of some dependencies using legacy paths.
I think the easiest is to just remove them and check if it works :man_shrugging:

**Comment by htdvisser** [comment]
Removed them, still works. However, there were issues with `golang.org/x` modules, so I had to add those 🤦‍♂

Also, we now have test failures in

- [ ] pkg/applicationserver (NATS)
- [ ] pkg/applicationserver/io/mqtt
- [ ] pkg/gatewayserver/io/mqtt
- [ ] pkg/web

@johanstokking / @adriansmares can you look at the first three, then I'll look at the last one

**Comment by adriansmares** [comment]
#883 fixes the NATS test by being compatible with the latest version of `go-cloud.dev`.

**Comment by johanstokking** [comment]
Fixed MQTT tests, now compatible with latest tagged release of Paho

**Comment by johanstokking** [comment]
Blocked by #883

**Comment by johanstokking** [comment]
@adriansmares can you rebase and confirm that `pkg/applicationserver` passes tests again?

**Comment by adriansmares** [comment]
Rebased on top of the latest master. Tests succed now, only the `web` package one is still not working.

**Comment by htdvisser** [comment]
@rvolosatovs 	@johanstokking 	Travis is satisfied. Please approve.

File: go.mod [hunk 1]
```diff
@@ -4,102 +4,110 @@ replace github.com/grpc-ecosystem/grpc-gateway => github.com/TheThingsIndustries
 
 replace github.com/robertkrimen/otto => github.com/TheThingsIndustries/otto v0.0.0-20181129100957-6ddbbb60554a
 
-replace github.com/labstack/echo/v4 => github.com/TheThingsIndustries/echo/v4 v4.0.1-0.20190409124425-ee570f243713
+// Fix versions of golang.org/x modules
```
Comment (rvolosatovs) [inline go.mod]: It's quite clear that you `Fix versions of golang.org/x modules` here, can you add reasoning *why* you we need these replacements?

File: pkg/gatewayserver/io/mqtt/mqtt_test.go [hunk 2]
```diff
@@ -93,8 +93,7 @@ func TestAuthentication(t *testing.T) {
 					client.Disconnect(uint(timeout / time.Millisecond))
 				}
 			} else {
-				a.So(ok, should.BeTrue)
```
Comment (rvolosatovs) [inline pkg/gatewayserver/io/mqtt/mqtt_test.go]: Why did this change?

File: pkg/applicationserver/io/mqtt/mqtt_test.go [hunk 3]
```diff
@@ -115,8 +115,7 @@ func TestAuthentication(t *testing.T) {
 					client.Disconnect(uint(timeout / time.Millisecond))
 				}
 			} else {
-				a.So(ok, should.BeTrue)
```
Comment (rvolosatovs) [inline pkg/applicationserver/io/mqtt/mqtt_test.go]: IDEM

File: pkg/gatewayserver/io/mqtt/mqtt_test.go (hunk 2 above)
Comment (johanstokking) [reply pkg/gatewayserver/io/mqtt/mqtt_test.go]: This is a breaking change in Paho.

Formerly, on timeout, `true` was returned.

Now, the return value indicates whether the action was successful (i.e. `false` on timeout). So in this statement, we assert `false` as we expect a timeout when the test case is supposed to fail.

See https://github.com/eclipse/paho.mqtt.golang/commit/ba971f185f1bce82b0e8d19c74da95e37c68234a#diff-ac704b2e102a5731d9d1b1707d3bc7a1R80

File: pkg/applicationserver/io/mqtt/mqtt_test.go (hunk 3 above)
Comment (johanstokking) [reply pkg/applicationserver/io/mqtt/mqtt_test.go]: See above

**Pull request merged**